### DISCUSSION TITLE
Run a script when moving/dragging the note to another level, parent or child (scripting)

### DISCUSSION THREAD

**Discussion by nicollasricas** [TriliumNext/Trilium#2898] [Q&A] [answered]
Is there a way to run a script when moving/dragging the note to another level, parent or child?

`~runOnNoteChange` - Doesn't work

Example:
Run after I move the **Fix Z** note inside the **Done** note.

![image](https://user-images.githubusercontent.com/7860985/172001862-036eb34e-ae6d-4068-b4da-b842d2e0fd5f.png)

**Comment by zadam** [comment, marked as answer]
Hi, I just added two new [events](https://github.com/zadam/trilium/wiki/Events):

* `runOnBranchCreation` - executes when a branch is created. Branch is a link between parent note and child note and is created e.g. when cloning or moving note.
* `runOnBranchDeletion` - executes when a branch is delete. Branch is a link between parent note and child note and is deleted e.g. when moving note (old branch/link is deleted).

Move is represented by deleting the old branch (representing old location) and creating a new branch (new target location).

This is present in just released 0.52.1-beta

**Comment by nicollasricas** [comment]
Work like a charm. Thanks for the update!
